src/TiledIteration.jl

```diff
@@ -64,10 +64,11 @@ function Base.iterate(iter::EdgeIterator, state)
 end
 
 @inline function nextedgeitem(iter::EdgeIterator, I::CartesianIndex)
-    !(I ∈ iter.inner) && return I
-    state = (last(iter.inner)[1], tail(I.I)...)
-    newI = CartesianIndex(_inc(state, iter.outer))
-    nextedgeitem(iter, newI)
+    while I.I[1] == first(iter.inner)[1] && I ∈ iter.inner
```

Comment:
Hmmm, I'm a bit lost in the loop condition here. Would you mind adding a few comments to the code to explain this a bit?

Reply:
Yes of course. I added an explanation.